processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -511,6 +538,76 @@ private void applyParameterNameBasedMapping() {
             }
         }
 
+
+        private ConstructorMapping createConstructorMapping() {
+            Constructor constructor = method.getResultType().getConstructor();
+
+            if (constructor != null) {
+                LinkedHashMap<String, PropertyMapping> mappings = new LinkedHashMap<String, PropertyMapping>(  );
+
+                for (String propertyName : constructor.getPropertyNames()) {
+                    if ( constructorPropertyMappings.containsKey( propertyName ) ) {
+                        mappings.put( propertyName, constructorPropertyMappings.get( propertyName ) );
+                    }
+                    else {
+                        Parameter parameter = constructor.getParameter( propertyName );
+                        PropertyMapping nullMapping = new ConstantMappingBuilder()
+                                .mappingContext( ctx )
+                                .sourceMethod( method )
+                                .constantExpression( "\"" + getConstructorConstantExpression( parameter ) + "\"" )
+                                .targetProperty( PropertyEntry.forTargetReference(
+                                        propertyName.split( "\\." ),
+                                        null,
+                                        constructor.getExecutableElement(),
+                                        parameter.getType() ) )
+                                .targetPropertyName( propertyName )
+                                .existingVariableNames( existingVariableNames )
+                                .build();
+
+                        mappings.put( propertyName, nullMapping );
+                    }
+                }
+
+                return new ConstructorMapping( constructor, mappings );
+            }
+
+            return null;
+        }
+
+        private String getConstructorConstantExpression(Parameter parameter) {
+            if ( parameter.getType().isPrimitive() ) {
+                TypeKind kind = parameter.getType().getTypeMirror().getKind();
+                if ( kind == TypeKind.BOOLEAN ) {
```

Comment:
Hmm. I just did a search on 0.0. Seems that we have similar code already in `Type` and `IterableMappingMethod`.. Perhaps already a refactoring point :smile: